runway/blueprints/staticsite/templates/replicated_lambda_remover.template.py

```diff
@@ -0,0 +1,67 @@
+"""Replicated Lambda Remover State Machine."""
+from typing import Any, Dict  # pylint: disable=unused-import
+
+import boto3
+
+
+def handler(event,  # type: Dict
+            _context  # type: Dict
+           ):  # noqa: E124
+    # type: (...) -> Dict[str, Any]
+    """State Machine step to attempt deletion of replicated Lambdas.
+
+    Given a list of Arns will go through each attempting to
+    delete. If the resource is not found it will move to the
+    next arn, assuming that it has already been deleted. All
+    other errors will move the step function to a halt step
+    where it will then re-request the deletion process.
+
+    Args:
+        event (dict): The AWS Step Function execution event
+    """
+    arns = event.get('FunctionArns')
+    self_destruct = event.get('SelfDestruct')
+    lambda_client = boto3.client('lambda')
+
+    # End early if we don't have any arns
+    if not arns:
+        return {
+            'SelfDestruct': self_destruct,
+            'FunctionArns': arns,
+            'deleted': True,
+            'status': 'No Arns To Delete'
+        }
+
+    deletions = []
+
+    try:
+        # Loop through all arns passed in
+        for _ in range(0, len(arns)):
+            try:
+                deletions = [
+                    lambda_client.delete_function(FunctionName=arn) for arn in arns
+                ]
```

Comment:
I think this loop logic can be reduced slightly, something like:

```
        deletions = []
        for arn in arns:
            try:
                deletions.append(lambda_client.delete_function(FunctionName=arn))
```

Reply:
From other PRs I've had I've been told that using comprehensions is the more "pythonic" way of handling these loops. I'm happy to change this if you'd like, but I'm beginning to be unsure in that case when the right time to use them is? Would you be able to give me some pointers in that regard so I make the correct decisions going forward?

Reply:
It _feels_ like there should be a way to do this without both the `for` loop and the list comprehension (intent isn't to try the deletions `len(arns) * len(arns)` times, right?)

Reply:
D'oh. Ok I see what you're saying now. I'll get this fixed.

Reply:
Updated... finally